### ISSUE TITLE
Replace deprecated TextField component with TextField from the component-library

### ISSUE THREAD

**Issue by georgewrmarshall** [MetaMask/metamask-extension#20483]
### Description

Currently, the extension is using an outdated `TextField` component, which needs to be replaced with the new `TextField` component.

This is a massive undertaking by itself and creating a single PR would be too large. Smaller PRs can be submitted against this issue to ensure easier review and gradual improvements. It also means that this issue will not be assigned to a single engineer

### Technical Details

- Replace instances of `TextField` component (`ui/components/ui/text-field/text-field.component.js`) with `TextField` component (`ui/components/component-library/text-field/text-field.js`)
- Component APIs are slightly different so ensure all props have been migrated appropriately
- Ensure there are no visual or functional regressions

### Acceptance Criteria

- Instances of `TextField` component are completely replaced with the new `TextField` component
- The component APIs are updated to reflect the changes in the new `TextField`` component and there is no functional changes or visual regression
- Each Pull Request (PR) should include **no more than 3 files**
- The code changes should pass Jest tests, linting, and Storybook without any errors.
- The PR must include before and after screenshots of the UI to ensure there are no visual regressions.

If the acceptance criteria is not met, PRs may be closed.

### Difficulty: Intermediate

Good first issue for: External contributors who are familiar with running the extension locally, have knowledge of React, component props, Jest tests, linting, and Storybook, and want to contribute to improving the cohesiveness of UI in the extension

**Comment by ShatilKhan** [comment]
Hi @georgewrmarshall 
I would like to take on this issue. I've recently done some contributions into components in meshery.io. 
Can you please assign me this issue?

**Comment by subhajit20** [comment]
@georgewrmarshall Made a PR.
Can you assign it to me?

**Comment by mohdshubair313** [comment]
Hey @georgewrmarshall **can you assign this issue to me**. I can work on this issue @georgewrmarshall

**Comment by besalraj30** [comment]
Hi

I’d like to work on this  issue. Could you assign it to me?  @georgewrmarshall

**Comment by hash88o** [comment]
Hi @georgewrmarshall,

I'd like to contribute to this TextField migration effort. I have experience with React, component migrations, Jest testing, and Storybook documentation, and I'm comfortable working with the MetaMask extension codebase.

Before I start, I wanted to confirm:
1. **Are there specific files or directories that still need migration?** I want to ensure I'm working on files that haven't been claimed by other contributors.
2. Should I comment here with my chosen files before starting to avoid any duplicate work?

I plan to start with a small PR (2-3 files maximum) and will ensure:
- Proper migration from the old `TextField` (`ui/components/ui/text-field/text-field.component.js`) to the new one (`ui/components/component-library/text-field/text-field.js`)
- All props are correctly updated to match the new API
- Complete test coverage with passing Jest tests
- No visual or functional regressions (with before/after screenshots)
- Clean linting and Storybook documentation

I've successfully set up the extension locally and am ready to contribute. Looking forward to helping improve UI consistency across the extension!

Thanks!

**Comment by georgewrmarshall** [comment]
Hey @hash88o, thanks for reaching out and for your interest in contributing to MetaMask! 🙌

It sounds like you’re all set up and ready to go. I did a quick search for remaining deprecated `TextField` components and found a few instances you could tackle.

*These deprecated `TextField` instances can be replaced with `TextField` from the component-library:*

* [create-new-vault.js](https://github.com/MetaMask/metamask-extension/blob/f35e29d7ac97a1b42a6bb1e10162d98c9157e837/ui/components/app/create-new-vault/create-new-vault.js#L105-L128)
* [customize-nonce.component.js](https://github.com/MetaMask/metamask-extension/blob/f35e29d7ac97a1b42a6bb1e10162d98c9157e837/ui/components/app/modals/customize-nonce/customize-nonce.component.js#L108-L122)
* [edit-approval-permission.component.js](https://github.com/MetaMask/metamask-extension/blob/f35e29d7ac97a1b42a6bb1e10162d98c9157e837/ui/components/app/modals/edit-approval-permission/edit-approval-permission.component.js#L172-L187)
* [advanced-tab.component.js](https://github.com/MetaMask/metamask-extension/blob/f35e29d7ac97a1b42a6bb1e10162d98c9157e837/ui/pages/settings/advanced-tab/advanced-tab.component.js#L519-L529)
* [add-contact.component.js](https://github.com/MetaMask/metamask-extension/blob/f35e29d7ac97a1b42a6bb1e10162d98c9157e837/ui/pages/settings/contact-list-tab/add-contact/add-contact.component.js#L157-L166)
* [edit-contact.component.js](https://github.com/MetaMask/metamask-extension/blob/f35e29d7ac97a1b42a6bb1e10162d98c9157e837/ui/pages/settings/contact-list-tab/edit-contact/edit-contact.component.js#L131-L178)
* [security-tab.component.js](https://github.com/MetaMask/metamask-extension/blob/f35e29d7ac97a1b42a6bb1e10162d98c9157e837/ui/pages/settings/security-tab/security-tab.component.js#L742-L749)

*This deprecated `TextField` instance can be replaced with `TextFieldSearch`:*

* [settings-search.js](https://github.com/MetaMask/metamask-extension/blob/f35e29d7ac97a1b42a6bb1e10162d98c9157e837/ui/pages/settings/settings-search/settings-search.js#L94-L107)

Feel free to pick one or more of these and open a PR. I’d recommend starting with a single replacement first so we can review and confirm the approach before you tackle the rest.

Thanks again for helping improve MetaMask! 💪

**Comment by hash88o** [comment]
Thanks so much for the guidance, @georgewrmarshall! 

I'll start with a single file as you recommended to ensure my approach aligns with the team's expectations. I'll begin with **`create-new-vault.js`** and submit a PR once that PR is reviewed and the approach is confirmed, I'll be happy to tackle more files from the list.

I'll comment here again once the PR is ready for review. Thanks for the opportunity to contribute to MetaMask!

**Comment by hash88o** [comment]
Hi @georgewrmarshall,

I've submitted a PR for the `create-new-vault.js` migration: #36980

The PR includes:
- Migration from deprecated TextField to FormTextField
- Updated prop mappings (inputProps, error as boolean, helpText)
- All local tests passing

Would appreciate your review when you get a chance. Thanks! 🙏

**Comment by LeonxLJX** [comment]
I'd like to take this one — I'll follow up with a PR. (claiming via @LeonxLJX)

**Comment by LeonxLJX** [comment]
Hi! I'd like to replace deprecated TextField with the new component. Plan: swap, test. May I be assigned?

**Comment by georgewrmarshall** [comment]
Closing as no longer needed, thanks for all the contributions!
